File: app/src/main/java/com/ntsoftware/vspc/myvspc/screens/discover/holders/TodayHolder.java
package com.ntsoftware.vspc.myvspc.screens.discover.holders;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.ntsoftware.vspc.myvspc.screens.discover.holders.base.DiscoverHolder;
import com.ntsoftware.vspc.myvspc.R;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class TodayHolder extends DiscoverHolder {


    private TextView info_text;

    public TodayHolder(@NonNull View itemView) {
        super(itemView);
        info_text = itemView.findViewById(R.id.tv_home_info_date);
    }

    public void bind() {
        Locale loc = itemView.getContext().getResources().getConfiguration().locale;
        info_text.setText(new SimpleDateFormat("EEEE, dd MMMM",loc).format(new Date()));
    }
}

File: app/src/main/java/com/ntsoftware/vspc/myvspc/screens/discover/holders/TodayHolder.kt
package com.ntsoftware.vspc.myvspc.screens.discover.holders

import android.view.TextureView
import android.view.View
import android.widget.TextView
import com.ntsoftware.vspc.myvspc.R
import com.ntsoftware.vspc.myvspc.screens.discover.holders.base.DiscoverHolder
import java.text.SimpleDateFormat
import java.util.*

class TodayHolder(itemView: View): DiscoverHolder(itemView) {

    private val info_text: TextView = itemView.findViewById(R.id.tv_home_info_date)

    override fun bind() {
        val locale = itemView.context.resources.configuration.locale
        info_text.text = SimpleDateFormat("EEEE, dd MMMM",locale).format(Date()).capitalize()
    }
}